test/animate.spec.js

```diff
@@ -1,278 +1,330 @@
-import {VelocityAnimator} from '../src/animator';
-import {animationEvent} from 'aurelia-templating';
-import {initialize} from 'aurelia-pal-browser';
+import { VelocityAnimator } from '../src/animator';
+import { animationEvent } from 'aurelia-templating';
+import { initialize } from 'aurelia-pal-browser';
 
 jasmine.getFixtures().fixturesPath = 'base/test/fixtures/';
 
-describe('the animate method', () => {
-  let elem;
-  let animator;
+describe('`animate` function', () => {
   let container;
+  let animator;
+  let elemGroup;
+  let elem;
 
-  beforeAll(() => initialize());
-
-  beforeEach(() => {
-    //stop all animations running on the test element
-
+  // -------- SETUP
+  beforeAll(() => {
+    initialize();
     loadFixtures('animation.html');
-    elem = $('#test-simple').eq(0)[0];
+    elem = $('#test-animate').eq(0)[0];
+    elemGroup = $('.test-animate-group');
     container = $('#animation').eq(0)[0];
     animator = new VelocityAnimator(container);
   });
 
+  beforeEach(() => {
+    // stop all animations
+    animator.stop(elem, true); // stop all animations
+
+    // clear all styles
+    elem.removeAttribute('style');
+    for (const el of Array.from(elemGroup)) el.removeAttribute('style');
+  });
+
+  // -------- TESTS: Main
   it('returns a promise', () => {
-    let result = animator.animate(elem, 'fadeIn');
-    expect(result.then).toBeDefined();
+    const result = animator.animate(elem, 'fadeIn');
+    expect(result instanceof Promise).toBe(true);
```

Comment:
this might not work on all environments and only as long as we use native Promises, which in our case though shouldn't be a biggy. Checking for the presence of then used to be the goto way, but that is already a loong time ago